api/src/endpoints/neo4j.js

```diff
@@ -12,65 +12,48 @@ import {
   getRelatedReactionsForSubsystem,
   getRelatedMetabolites,
   getRandomComponents,
-  modelSearch,
-  globalSearch,
   getInteractionPartners,
   getMapsListing,
   mapSearch,
+  search,
 } from 'neo4j/index';
 
 const neo4jRoutes = express.Router();
 
-const fetchWith = async (req, res, getter) => {
-  const { id, version } = req.params;
-  const { limit, model, full, searchTerm } = req.query;
+const fetchWith = async (req, res, queryHandler) => {
+  const { id } = req.params;
+  const { model, version, limit, full, searchTerm } = req.query;
 
   try {
-    const result = await getter({ id, version, limit, model, full, searchTerm });
+    const result = await queryHandler({ id, version, limit, model, full, searchTerm });
     res.json(result);
   } catch (e) {
     res.status(400).send(e);
   }
 };
 
-neo4jRoutes.get('/:version/compartments/:id', async (req, res) => fetchWith(req, res, getCompartment));
-neo4jRoutes.get('/:version/compartments/:id/related-reactions', async (req, res) => fetchWith(req, res, getRelatedReactionsForCompartment));
+neo4jRoutes.get('/compartments/:id', async (req, res) => fetchWith(req, res, getCompartment));
+neo4jRoutes.get('/compartments/:id/related-reactions', async (req, res) => fetchWith(req, res, getRelatedReactionsForCompartment));
 
-neo4jRoutes.get('/:version/genes/:id', async (req, res) => fetchWith(req, res, getGene));
-neo4jRoutes.get('/:version/genes/:id/related-reactions', async (req, res) => fetchWith(req, res, getRelatedReactionsForGene));
+neo4jRoutes.get('/genes/:id', async (req, res) => fetchWith(req, res, getGene));
+neo4jRoutes.get('/genes/:id/related-reactions', async (req, res) => fetchWith(req, res, getRelatedReactionsForGene));
```

Comment:
For https://github.com/MetabolicAtlas/MetabolicAtlas/pull/508#discussion_r481796918: 

All of the endpoints would look similar in terms of query params. In this case it would be `/genes/:id/related-reactions?model=HumanGem`

While query params may not be the best choice for some endpoints for now, I'm still highly in favour of them for the the consistency and flexibility for future use cases.

Reply:
Sounds good!